src/SimulationStrategyFactory.py

```diff
@@ -0,0 +1,82 @@
+from enum import IntEnum
+from src.LossyBosonSamplingExactDistributionCalculators import BosonSamplingExperimentConfiguration
+from src.simulation_strategies.SimulationStrategy import SimulationStrategy
+from src.simulation_strategies.FixedLossSimulationStrategy import FixedLossSimulationStrategy
+from src.simulation_strategies.UniformLossSimulationStrategy import UniformLossSimulationStrategy
+from src.simulation_strategies.GeneralizedCliffordsSimulationStrategy import GeneralizedCliffordsSimulationStrategy
+from src.simulation_strategies.CliffordsRSimulationStrategy import CliffordsRSimulationStrategy
+
+
+class StrategyTypes(IntEnum):
+    FIXED_LOSS = 1
+    UNIFORM_LOSS = 2
+    CLIFFORD_R = 3
+    GENERALIZED_CLIFFORD = 4
+
+
+class SimulationStrategyFactory:
+    def __init__(self, experiment_configuration: BosonSamplingExperimentConfiguration,
+                 strategy_type: StrategyTypes = StrategyTypes.FIXED_LOSS):
+        self._experiment_configuration = experiment_configuration
+        self._strategy_type = strategy_type
+
+    def set_strategy_type(self, strategy_type: StrategyTypes) -> None:
+        self._strategy_type = strategy_type
+
+    def set_experiment_configuration(self, experiment_configuration: BosonSamplingExperimentConfiguration) -> None:
+        self._experiment_configuration = experiment_configuration
+
+    def generate_strategy(self) -> SimulationStrategy:
+        """
+            Generates simulation strategy of desired type. The type is selected in the constructor.
+        :return: Simulation strategy of desired type.
+        """
+        if self._strategy_type == StrategyTypes.UNIFORM_LOSS:
```

Comment:
Nitpick: How about a dictionary in the `__init__` in form of
```
self._strategy_mapping = {
    StrategyTypes.CLIFFORD_R: self.__generate_r_cliffords_strategy
}
```
and replacing this function with
```
handler = self._strategy_mapping.get(self._strategy_type, self.__generate_fixed_losses_strategy)
return handler()
```
?

Reply:
That's elegant, I like it a lot.